src/renderer/components/Editor/SquareBracket.tsx

```diff
@@ -0,0 +1,45 @@
+import { Box } from '@mui/system';
+import { useState } from 'react';
+import colors from 'renderer/colors';
+import SquareBracketHover from './SquareBracketHover';
+
+interface Props {
+  isLast: boolean;
+  isTakeGroupOpened: boolean;
+}
+
+const SquareBracket = ({ isLast, isTakeGroupOpened }: Props) => {
+  const bottomWidth = isLast || !isTakeGroupOpened ? '2px' : '0px';
+
+  const [isHovering, setIsHovering] = useState(false);
+
+  const handleMouseOver = () => {
```

Comment:
If you're not going to memoise them I think it's fine to just inline them since you only use them once

i.e. `onMouseOver={() => setIsHovering(true)}`

Reply:
I've done this in https://github.com/chloebrett/mlvet/pull/284/commits/a473d51cd03cedd7a5a0d82c563dabe7052d6deb